lib/remoteproc/remoteproc_virtio.c

```diff
@@ -40,6 +40,9 @@ static unsigned char rproc_virtio_get_status(struct virtio_device *vdev)
 	rpvdev = metal_container_of(vdev, struct remoteproc_virtio, vdev);
 	vdev_rsc = rpvdev->vdev_rsc;
 	io = rpvdev->vdev_rsc_io;
+
+	RSC_TABLE_INVALIDATE(vdev_rsc, sizeof(vdev_rsc));
```

Comment:
I don't think this is correct either.

This is expanding to:
```
struct fw_rsc_vdev *vdev_rsc;

metal_cache_flush(&vdev_rsc, sizeof(vdev_rsc))
```
That is you are flushing the pointer, not the actual table.

The correct expansion should be;
```
metal_cache_flush(vdev_rsc, sizeof(*vdev_rsc));
```

Reply:
Right. 
I added a new version fixing everything.